File: ConvexCodeExamplesExternal/src/main/Python/amqp/publishC114.py
#!/usr/bin/env python
import pika
import ssl

ssl_context = ssl.create_default_context()
ssl_options = pika.SSLOptions(ssl_context, "amqp.convexglobal.io")

parameters = pika.URLParameters('amqps://01nI9pMLYRL4BWTY:guest@amqp.convexglobal.io:5671/stream')

#credentials = pika.PlainCredentials('01nI9pMLYRL4BWTY', '01nI9pMLYRL4BWTY')
#parameters = pika.ConnectionParameters(port=5671, virtual_host='stream', credentials=credentials, ssl_options=ssl_options)

connection = pika.BlockingConnection(parameters)
channel = connection.channel()

channel.queue_declare(queue='2cc2aa')

channel.basic_publish(exchange='2cc2aa', routing_key='', body='Hello World!')
print(" [x] Sent 'Hello World!'")
connection.close()

File: ConvexCodeExamplesExternal/src/main/JS/amqp/publishC114.js
//Simple Node-based example to publish data using AMQP
//Libraries required:

var dataResourceId = '2cc2aa';
var accessToken = '01OZLElajKpQJonw';

var amqp = require('amqplib/callback_api');

amqp.connect('amqps://' +accessToken +'@amqp.convexglobal.io:5671/stream', function(error0, connection) {
  if (error0) {
    throw error0;
  }
  connection.createChannel(function(error1, channel) {
    if (error1) {
      throw error1;
    }
    var queue = '2cc2aa';
    var msg = 'Hello world';
   
    channel.publish(queue, queue, Buffer.from(msg));

    console.log(" [x] Sent %s", msg);
  });
});